bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,317 @@
+            raise TypeError(f"Argument is a {type(sentence)} instead of a str.")
+        return sentence.strip(string.punctuation).split()[-1]
+
+    @memoize
+    async def _get_rhyme_set(
+        self,
+        word: str,
+        near_rhyme_min_score: int = 2000
+    ) -> None:
+        """
+        Accesses web APIs to get rhymes and returns a set.
+
+        `near_rhyme_min_score` is to filter out near rhymes that barely rhyme
+        with the word. The equivalent for exact rhymes is not necessary as they
+        already rhyme.
+
+        Should additional web APIs for rhymes be added, the `min_score` needs
+        to be tuned. Perhaps it should be added as an element to the tuple of
+        the element of the `self.rhyme_websites` list.
+        """
+        rhyme_set = set()
+
+        for is_exact, website in self.rhyme_websites:
+            min_score = 0 if is_exact else near_rhyme_min_score
+
+            async with self.bot.http_session.get(
+                website + word,
+                timeout=10
+            ) as response:
+                curr_set = set(
+                    data["word"] for data in await response.json()
+                    if data.get("score", 0) >= min_score
+                )
+                rhyme_set |= curr_set
+
+        return rhyme_set
+
+    async def _get_rhyming_line(
+        self,
+        word_rhymes: List[str],
+        limiter: int = 80000
+    ) -> str:
+        """
+        Returns a sentence string with a last word in `word_rhymes`.
+
+        The function will continue to iterate through sentences provided by the
+        markov model, until it finds one that rhymes or until it reaches the
+        limiter.
+        """
+        curr = 0
+        line = self.model.make_short_sentence(random.randint(50, 120))
+        while self._get_last_word(line) not in word_rhymes:
+            if curr >= limiter:
+                raise RhymingSentenceNotFound
+
+            line = self.model.make_short_sentence(random.randint(50, 120))
+            curr += 1
+
+        return line
+
+    def _get_unit_count(self, scheme: str) -> Dict[str, bool]:
+        """Checks how many times a unit occurs in a rhyme scheme."""
+        return {char: scheme.count(char) for char in set(scheme)}
+
+    async def _init_unit(
+        self,
+        is_last_line: bool,
+        rhyme_track: Dict[str, Set[str]],
+        unit: str
+    ) -> str:
+        rhyme_set = set()
+        while len(rhyme_set) == 0:
+            line = self.model.make_short_sentence(
+                random.randint(50, 120)
+            )
+
+            rhyme_set = await self._get_rhyme_set(
+                instance=self,
+                word=self._get_last_word(line),
+                is_instant_fail=(not is_last_line)
+            )
+
+        rhyme_track[unit] = rhyme_set
+        return line
+
+    @commands.command()
+    async def poem(self, ctx: commands.Context, scheme: str) -> None:
```

Comment:
We can use the parameter `brief` and `help` of the `commands.command` here, since the current doc-string contains a lot information which the users don't need.

Reply:
I don't get what you mean by this.

Reply:
[Snippet from another project](https://github.com/gurkult/gurkbot/blob/f81de580509f65f60946b04f46aa252650639293/bot/exts/utils/eval.py#L53) Does this help?